From 070fe09663274242ae3af572f46e12ea2c7a95c6 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 16 Jun 2019 22:33:20 +0200
Subject: [PATCH 1/1] efi_loader: console incorrectly advertised left logo key

Avoid to signal that the left logo key is pressed, when it is not.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_console.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index 0ecc848e2e..81d3c7254d 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -499,16 +499,20 @@ void set_shift_mask(int mod, struct efi_key_state *key_state)
  */
 static int analyze_modifiers(struct efi_key_state *key_state)
 {
-	int c, mod = 0, ret = 0;
+	int c, mod, ret = 0;
 
 	c = getc();
 
 	if (c != ';') {
 		ret = c;
-		if (c == '~')
+		if (c == '~') {
+			/* no modifier */
+			mod = 1;
 			goto out;
+		}
 		c = getc();
 	}
+	mod = 0;
 	for (;;) {
 		switch (c) {
 		case '0'...'9':
-- 
2.20.1

